src/core/net/dns_client.cpp

```diff
@@ -792,7 +792,7 @@ Error Client::Start(void)
 {
     Error error;
 
-    SuccessOrExit(error = mSocket.Open(Ip6::kNetifUnspecified));
+    SuccessOrExit(error = mSocket.Open(Ip6::kNetifThreadInternal));
```

Comment:
From what I recall, there was a reason or specific use case for this.

`Dns::Client` is designed to be general-purpose. For example, `otDnsConfig` allows a user to specify the address of the server to query. The server may be another software entity on the same device, so we want to keep `kNetif` as flexible as possible and allow traffic to go through the platform.